components/renku_data_services/session/db.py

```diff
@@ -249,14 +261,27 @@ async def update_launcher(
                         message=f"Session environment with id '{environment_id}' does not exist or you do not have access to it."  # noqa: E501
                     )
 
+            resource_class_id = kwargs.get("resource_class_id")
+            if resource_class_id is not None:
+                res = await session.scalars(
+                    select(schemas.ResourceClassORM).where(schemas.ResourceClassORM.id == resource_class_id)
+                )
+                resource_class = res.one_or_none()
+                if resource_class is None:
+                    raise errors.MissingResourceError(
+                        message=f"Resource class with id '{resource_class_id}' does not exist or you do not have access to it."  # noqa: E501
+                    )
```

Comment:
issue: same here, we need to properly validate that a user actually has access to the class.